### DISCUSSION TITLE
Problem trying Create Seurat Object

### DISCUSSION THREAD

**Discussion by FernandaScar** [satijalab/seurat#5982] [Q&A] [answered]
Hello everyone!

I'm new at Single Cell, sorry if my question is such a basic thing to solve.
I'm trying to create an seurat object to perform DE analysis in Rstudio using an public data available on GEO GSE93374. They don't have the barcodes, features and matrix in separate files, so I try to use the first file on sup.files GSE93374_Merged_all_020816_BatchCorrected_LNtransformed_doubletsremoved_Data.txt.gz.
The data is already in 10x I supposed, so I used this commands:

CreateSeuratObject(counts = "here I put the file", project = "SeuratObject", assay = "RNA", names.field = 1, names.delim = "_", meta.data = NULL)

I also try something like this:
SeuratObject <- CreateSeuratObject(counts = counts, min.cells = 3, min.features = 0, assay  = "RNA")

I both cases, the Seurat gives me the same error: Error in CreateSeuratObject(counts = counts, min.cells = min.cells, min.features = min.features) : **No cell names (colnames) names present in the input matrix.**

I used data.table and the fread command to look the data (because i't's a pretty large file) and my colnames are the barcodes, which they are in the file.
I already try to search this problem online but none of the suggestions worked for me. I try use data$'Gene Expression' and also colnames(counts) <- paste0(colnames(counts) 1:ncol(counts)).

**Someone had the same problem? What can I do about it?**

Thankx in advance!

**Comment by samuel-marsh** [comment, marked as answer]
Hi,

So in taking look at things the data is not 10X data as the study says it is dropseq but that's ok.  The only function that requires data to be in 10X format are the `Read10X` functions.  The only thing that `CreateSeuratObject` requires is data in a data.frame or matrix where the rows are the features/genes and the columns are the cells.  The `Read10X` functions simply take 10X outputted data and create this so it can be used by `CreateSeuratObject`.

In this case we can do the following:
```
library(Seurat)
library(tibble)
library(magrittr)

# Read in data
data <- data.table::fread("~/Downloads/GSE93374_Merged_all_020816_BatchCorrected_LNtransformed_doubletsremoved_Data.txt.gz", data.table = F)

# Gene names are in the first column so we need to move them to rownames
data <- data %>% 
  column_to_rownames("V1")

# Create Seurat
seurat <- CreateSeuratObject(counts = data)
```

Be aware though that similar to the other issue you commented on this data is not raw expression data but normalized data.  As such do not run any further normalization on this data (`NormalizeData` or `scTransform`).  Hopefully there are enough details in the paper/GEO to tell you how the data was normalized/corrected so you can accurately interpret the data.

Best,
Sam

**Reply by FernandaScar** [reply]
Hi Sam!

Thank you for you help!
I've try those commands and I think it was working out, but this error appeared: **Error in asMethod(object) : 
  'R_Realloc' could not re-allocate memory (3304504368 bytes)**

The file is large and my computer doesn't have a big RAM memory. Is this the issue why it's does not work this time? There is anything that I can do to solve and run this in my computer?

**Reply by samuel-marsh** [reply]
Hi,

Yes unfortunately that is simply going to be limitation of the RAM on your system.  Only way around it would be to use another system with more memory unfortunately.

Best.
Sam

**Reply by FernandaScar** [reply]
Thank you for your help anyway!!
